### PR TITLE
Allow optional commands in dcos-diagnostics

### PR DESCRIPTION
**Pull request by janisz**
## High-level description

Bump dcos-diagnostics to allow optional commands in dcos-diagnostics and configure number of diagnostics fetchers.

It's not a user facing change. It's just change how w handle optional commands in dcos diagnostics config and how we report errors.

## Corresponding DC/OS tickets (obligatory)

These DC/OS JIRA ticket(s) must be updated (ideally closed) in the moment this PR lands:

  - [DCOS_OSS-5026
](https://jira.mesosphere.com/browse/DCOS_OSS-5026)
  - [DCOS-51483](https://jira.mesosphere.com/browse/DCOS-51483)

## Related tickets (optional)

Other tickets related to this change:

  - [DCOS_OSS-<number>](https://jira.mesosphere.com/browse/DCOS_OSS-<number>) Foo the Bar so it stops Bazzing.

## Checklist for all PRs

  - [ ] Added a comprehensible changelog entry to `CHANGES.md` or explain why this is not a user-facing change:
  - [x] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [x] Change log from the last version integrated (this should be a link to commits for easy verification and review): [fc8aaeb0f15226...1389e1499b0](https://github.com/dcos/dcos-diagnostics/compare/fc8aaeb0f15226c4fabe2be18c94c72496739be3...1389e1499b0790a9bdf041514fe173a395a4b7b8)
  - [x] Test Results: [link to CI job test results for component](https://travis-ci.org/dcos/dcos-diagnostics/builds/515675122)
  - [x] Code Coverage (if available): [link to code coverage report](https://codecov.io/gh/dcos/dcos-diagnostics/branch/master/commits)
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and review process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using GitHub's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

File: packages/dcos-diagnostics/buildinfo.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "single_source" : {
     "kind": "git",
     "git": "https://github.com/dcos/dcos-diagnostics.git",
-    "ref": "20bc52c0e6521295f0332e4d91da98bd16c05903",
+    "ref": "99a19c1d7364d1464505ff9b3a363f40d173210c",
```
Comment (makkes) [inline packages/dcos-diagnostics/buildinfo.json]: This is actually the wrong hash. Current master is at `84f0d569086622d9c06c4f988100c21c5b474cd3` while `99a19c1d7364d1464505ff9b3a363f40d173210c` references a feature branch:
```
$ git branch -a --contains 99a19c1d7364d1464505ff9b3a363f40d173210c
  remotes/origin/allow_optional_commands
$
```

Comment (janisz) [reply packages/dcos-diagnostics/buildinfo.json]: Correct, I created this PR to check if tests are passing and they are not

Comment (makkes) [reply packages/dcos-diagnostics/buildinfo.json]: Ah, got it

**Comment by janisz** [comment]
@mesosphere-mergebot bump-ee

File: packages/dcos-diagnostics/buildinfo.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
   "single_source" : {
     "kind": "git",
     "git": "https://github.com/dcos/dcos-diagnostics.git",
-    "ref": "2f0d206619abbb083b6ac2b43efadf884a24769c",
+    "ref": "3734e2e03644449500427fb916289c4007dc5106",
```
Comment (makkes) [inline packages/dcos-diagnostics/buildinfo.json]: would you mind making this `1389e1499b0790a9bdf041514fe173a395a4b7b8` so my change is also in there?

**Comment by janisz** [comment]
@mesosphere-mergebot label Ready For Review
:wave: @drewvanstone @br-lewis @makkes  Can you review, please?

File: gen/dcos-config.yaml [hunk 3]
```diff
@@ -2822,10 +2829,6 @@ package:
           # are further forwarded into elasticsearch, where fields with leading
           # underscore are reserved for internal use.
           Strip_Underscores On
-          # Tail the journal (i.e., only send new logs). Without this option set
```
Comment (makkes) [inline gen/dcos-config.yaml]: Do you really want to remove that?

**Comment by jdef** [comment]
@mesosphere-mergebot label ship it

**Pull request merged**